packages/utilities-react/src/StepFlow/StepGroup.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect } from 'react';
+import { useStepContext } from '.';
+
+interface StepGroupProps {
+  children: React.ReactNode;
+}
+
+/**
+ * Container for steps. The children are filtered for only truthy values (via Children.toArray)
+ * and only the current step is returned from this component based on the `currentStep` value
+ * from `useStepContext`.
+ */
+export const StepGroup = ({ children }: StepGroupProps) => {
+  const { setTotalSteps, currentStep } = useStepContext();
+
+  const childrenArray = React.Children.toArray(children);
+  const childrenCount = React.Children.count(childrenArray);
+
+  // set total step count
+  useEffect(() => {
+    setTotalSteps(childrenCount);
+  }, [childrenCount, setTotalSteps]);
+
+  const currentStepComponent =
+    React.Children.toArray(childrenArray)[currentStep - 1];
```

Comment:
Was the first part of https://github.com/carbon-design-system/carbon/pull/20389#discussion_r2318675703 incorrect? I’m wondering why `React.Children.toArray(childrenArray)` is called if `childrenArray` should already be an array.